desktopexporter/internal/app/types/precise-timestamp.ts

```diff
@@ -0,0 +1,40 @@
+export class PreciseTimestamp {
+  constructor(
+    public milliseconds: number,
+    public nanoseconds: number
```

Comment:
Linking otel-js' HrTime: https://github.com/open-telemetry/opentelemetry-js/blob/f2cfd1327a5b131ea795301b10877291aac4e6f5/api/src/common/Time.ts#L30
https://github.com/open-telemetry/opentelemetry-js/blob/f2cfd1327a5b131ea795301b10877291aac4e6f5/packages/opentelemetry-core/src/common/time.ts

We might also simplify our lives by using a [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) internal representation but serializing to the 2 numbers on the wire. otel-js didn't do this due to needing to support legacy Node versions, now no longer an issue: https://github.com/open-telemetry/opentelemetry-js/issues/5680

otel-js is now also moving in this direction: https://github.com/open-telemetry/opentelemetry-js/pull/5522

Reply:
Super good call. I will do this in its own PR for my own mental health, since I will have to re-thread it through all the time/duration stuff, but it will simplify my life significantly.